src/ol/renderer/canvas/VectorLayer.js

```diff
@@ -315,7 +315,7 @@ class CanvasVectorLayerRenderer extends CanvasLayerRenderer {
         this.getLayer().hasListener(RenderEventType.PRERENDER) ||
         this.getLayer().hasListener(RenderEventType.POSTRENDER);
       if (!hasRenderListeners) {
-        return null;
+        return this.container;
```

Comment:
Maybe it makes sense to still return null when no background is set?
```suggestion
        return this.getLayer().getBackground() ? this.container : null;
```